library/spdm_crypt_lib/libspdm_crypt_asym.c

```diff
@@ -526,6 +526,172 @@ static bool libspdm_asym_func_need_hash(uint32_t base_asym_algo)
     return false;
 }
 
+/**
+ * libspdm_copy_signature_swap_endian_rsa
+ * Swaps the endianness of a RSA signature buffer. The ECDSA signature buffer is
+ * actually single internal buffer and can be swapped as a single whole buffer.
+ *
+ * There are two known usage models for buffers for this function.
+ * 1) Source and dest are the same (and their sizes).  This would be an in-place swap.
+ * 2) Source and dest are completely different (and dest size >= src size). No overlap.
+ *
+ * The case where source and dest are overlapped (and not the exact same buffer)
+ * is not allowed and guarded against with an assert.
+ **/
+static void libspdm_copy_signature_swap_endian_rsa(
+    uint8_t* dst,
+    size_t dst_size,
+    const uint8_t* src,
+    size_t src_size)
+{
+    /* RSA signature is a single buffer to be swapped */
+    size_t i;
+
+    if (src == dst) {
+        LIBSPDM_ASSERT(dst_size == src_size);
+
+        /* src and dst are same buffer. Swap in place. */
+
+        uint8_t byte;
+        for (i = 0; i < dst_size / 2; i++) {
+            byte = dst[i];
+            dst[i] = dst[dst_size - i - 1];
+            dst[dst_size - i - 1] = byte;
+        }
+    }
```

Comment:
```
}
else {
```
->
```
} else {
```
Lots of other instances as well. I'll see if I can configure `uncrustify` to catch that.

Reply:
@steven-bellock , I will file a new issue to catch format issue in `uncrustify`.

Reply:
will do.